Kinvey.Core/Store/PushRequest.cs

```diff
@@ -15,180 +15,360 @@
 
-			response = new PushDataStoreResponse<T>();
 		}
 
 		public override async Task <PushDataStoreResponse<T>> ExecuteAsync()
 		{
-			List<PendingWriteAction> pendingActions = SyncQueue.GetFirstN(limit, offset);
-
-			while (pendingActions != null && pendingActions.Count > 0)
-			{
-				var tasks = new List<Task<T>>();
-				foreach (PendingWriteAction pwa in pendingActions)
-				{
-						if (String.Equals("POST", pwa.action))
-						{
-							tasks.Add(HandlePushPOST(pwa));
-						}
-						else if (String.Equals("PUT", pwa.action))
-						{
-							tasks.Add(HandlePushPUT(pwa));
-						}
-						else if (String.Equals("DELETE", pwa.action))
-						{
-							tasks.Add(HandlePushDELETE(pwa));
-						}
-				}
-				try
-				{
-					await Task.WhenAll(tasks.ToArray());
-				}
-				catch (Exception e)
-				{
-					//Do nothing for now
-					response.AddKinveyException(new KinveyException(EnumErrorCategory.ERROR_DATASTORE_NETWORK,
-																	EnumErrorCode.ERROR_JSON_RESPONSE,
-																	"",
-																   e));  // TODO provide correct exception
-				}
-
-				List<T> resultEntities = new List<T>();
-				int resultCount = 0;
-				foreach (var t in tasks)
-				{
-					if (!EqualityComparer<T>.Default.Equals(t.Result, default(T)))
-					{
-						resultEntities.Add(t.Result);
-					}
-
-					resultCount++;
-				}
-
-				response.AddEntities(resultEntities);
-				response.PushCount += resultCount;
-
-				pendingActions = SyncQueue.GetFirstN(limit, offset);
-			}
-
-			return response;
+            var response = new PushDataStoreResponse<T>();
+
+            if (HelperMethods.IsLessThan(Client.ApiVersion, 5))
+            {
+                response = await PushSingleActionsAsync();
+            }
+            else
+            {
+                var pushMultiPostActionsResponse = await PushMultiPostActionsAsync();
+                response.SetResponse(pushMultiPostActionsResponse);
+
+                var pushSinglePutActionsResponse = await PushSingleActionsAsync("PUT");
+                response.SetResponse(pushSinglePutActionsResponse);
+
+                var pushSingleDeleteActionsResponse = await PushSingleActionsAsync("DELETE");
+                response.SetResponse(pushSingleDeleteActionsResponse);
+            }
+
+            return response;
 		}
 
 		public override Task<bool> Cancel()
 		{
 			throw new KinveyException(EnumErrorCategory.ERROR_GENERAL, EnumErrorCode.ERROR_METHOD_NOT_IMPLEMENTED, "Cancel method on PushRequest not implemented.");
 		}
 
-		private async Task<T> HandlePushPOST(PendingWriteAction pwa)
-		{
-			T entity = default(T);
-
-			try
-			{
-				int result = 0;
-
-				string tempID = pwa.entityId;
-
-				entity = Cache.FindByID(pwa.entityId);
+        #region Single actions pushes
+
+        private async Task<PushDataStoreResponse<T>> PushSingleActionsAsync(string action = null)
+        {
+            var response = new PushDataStoreResponse<T>();
+            var offset = 0;
+            var limit = 10;
```

Comment:
I believe this should be `100` instead of `10` since the multi-insert api can take up to 100 items per request

Reply:
@heyzooi 
Yes, the multi-insert api can take up to 100. But this variable is being used in the method “PushSingleActionsAsync” which has the old logic with single insert operations.